openapi_python_client/parser/responses.py

```diff
@@ -79,27 +80,30 @@ def empty_response(
     )
 
 
-def response_from_data(
+def response_from_data(  # noqa: PLR0911
     *,
     status_code: HTTPStatus,
     data: Union[oai.Response, oai.Reference],
     schemas: Schemas,
+    responses: Dict[str, Union[oai.Response, oai.Reference]],
     parent_name: str,
     config: Config,
 ) -> Tuple[Union[Response, ParseError], Schemas]:
     """Generate a Response from the OpenAPI dictionary representation of it"""
 
     response_name = f"response_{status_code}"
     if isinstance(data, oai.Reference):
-        return (
-            empty_response(
-                status_code=status_code,
-                response_name=response_name,
-                config=config,
-                data=data,
-            ),
-            schemas,
-        )
+        ref_path = parse_reference_path(data.ref)
+        if isinstance(ref_path, ParseError):
+            return ref_path, schemas
+        if not ref_path.startswith("/components/responses/"):
+            return ParseError(data=data, detail=f"$ref to {data.ref} not allowed in responses"), schemas
+        resp_data = responses.get(ref_path.split("/")[-1], None)
```

Comment:
I'm a little surprised that getting reference data is not a little more standardized or at least with a helper.... is doing a `split` and `-1` the general pattern in the rest of the repo?

Reply:
There's no use case in the rest of the repo for getting just the type name from a reference.

Reply:
Note that this isn't really "getting reference data". The helper is just converting the string `"#/components/schemas/MyTypeName"` to the string `"MyTypeName"`.

Reply:
Whoops, I was wrong, there are a couple other usages like this. So I've added a helper.